packages/core/admin/admin/src/pages/Settings/pages/ApiTokens/EditView/EditViewPage.tsx

```diff
@@ -12,77 +12,97 @@ import {
   useRBAC,
   useTracking,
 } from '@strapi/helper-plugin';
-import { Formik } from 'formik';
+import { AxiosError, AxiosResponse } from 'axios';
+import { Formik, FormikHelpers } from 'formik';
 import { useIntl } from 'react-intl';
 import { useQuery } from 'react-query';
 import { useSelector } from 'react-redux';
 import { useHistory, useRouteMatch } from 'react-router-dom';
 
-import { ApiTokenPermissionsProvider } from '../../../../../contexts/apiTokenPermissions';
+import {
+  ApiTokenPermissionsProvider,
+  PseudoEvent,
+} from '../../../../../contexts/apiTokenPermissions';
 import { selectAdminPermissions } from '../../../../../selectors';
 import { formatAPIErrors } from '../../../../../utils/formatAPIErrors';
 import { API_TOKEN_TYPE } from '../../../components/Tokens/constants';
+// @ts-expect-error not converted yet
 import FormHead from '../../../components/Tokens/FormHead';
+// @ts-expect-error not converted yet
 import TokenBox from '../../../components/Tokens/TokenBox';
 
-import FormApiTokenContainer from './components/FormApiTokenContainer';
-import LoadingView from './components/LoadingView';
-import Permissions from './components/Permissions';
-import init from './init';
-import reducer, { initialState } from './reducer';
-import { schema } from './utils';
+import { FormApiTokenContainer } from './components/FormApiTokenContainer';
+import { LoadingView } from './components/LoadingView';
+import { Permissions } from './components/Permissions';
+import { schema } from './constants';
+import { initialState, reducer } from './reducer';
+
+import type { Get, Update, Create } from '../../../../../../../shared/contracts/api-token';
+import type { List as ListContentApiPermissions } from '../../../../../../../shared/contracts/content-api/permissions';
+import type { List as ListContentApiRoutes } from '../../../../../../../shared/contracts/content-api/routes';
 
 const MSG_ERROR_NAME_TAKEN = 'Name already taken';
 
-const ApiTokenCreateView = () => {
+export const EditView = () => {
   useFocusWhenNavigate();
   const { formatMessage } = useIntl();
   const { lockApp, unlockApp } = useOverlayBlocker();
   const toggleNotification = useNotification();
   const history = useHistory();
   const permissions = useSelector(selectAdminPermissions);
-  const [apiToken, setApiToken] = useState(
+  const [apiToken, setApiToken] = React.useState<Get.Response['data']>(
+    // @ts-expect-error this is probably fine for now
     history.location.state?.apiToken.accessKey
       ? {
+          // @ts-expect-error this is probably fine for now
           ...history.location.state.apiToken,
         }
       : null
```

Comment:
I have to say this because i'll cry if i don't, this should use `useLocation` and pass a generic to declare what state is – you don't have to change it, but i have to say it